File: Problems/Algorithms/1351. Count Negative Numbers in a Sorted Matrix/count_negative.py
import bisect


class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        n, m = len(grid), len(grid[0])

        ans = 0
        for row in range(n):
            col = bisect.bisect_left(grid[row], 1, key=lambda x: -x)
            ans += m - col

        return ans

File: Problems/Algorithms/1351. Count Negative Numbers in a Sorted Matrix/CountNegative.kt
class Solution {
    fun countNegatives(grid: Array<IntArray>): Int {
        val n = grid.size
        val m = grid[0].size
        
        var r = n - 1
        var c = 0
        var ans = 0
        while (r >= 0 && c < m) {
            if (grid[r][c] < 0) {
                r--
                ans += m - c
            } else {
                c++
            }
        }
        
        return ans
    }
}

